app/admin/formwidgets/mapview/assets/js/mapview.shape.js

```diff
@@ -62,12 +62,34 @@
         drawOptions.visible = shape.default === 'circle'
 
         var drawing = new google.maps.Circle(drawOptions)
+        
```

Comment:
How about moving the logic for creating the info window and marker to maparea.js and only registering the mouseover and mouseout events here?

Reply:
Don't mind - though all the google.maps type logic is in mapview and mapview.shape rather than map area ?

Reply:
Right, move to `mapview.createShape` instead and remove the shape bagde code ;)

Reply:
No problem - done.

Reply:
I was wrong, your approach was better it doesn't look right in `mapview.createShape` lol. Also, would it be too much to use the distance unit setting instead of hardcoding the km?

Reply:
Not at all - thats done